pinot-core/src/test/java/org/apache/pinot/core/query/executor/MaxTasksExecutorTest.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.executor;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import org.testng.annotations.Test;
+
+import static org.testng.AssertJUnit.*;
+
+
+public class MaxTasksExecutorTest {
+
+  private static final int MAX_TASKS = 5;
+
+  @Test
+  public void testExecutor() {
+    MaxTasksExecutor ex = new MaxTasksExecutor(MAX_TASKS, Executors.newCachedThreadPool());
```

Comment:
Nit: I don't think declaring max thread in a constant is very useful. If more tests are added, they will probably not want to use the same number of threads. Given that this number is specific for each test, I think it is better to be explicit in the test to make it easier to read.

Reply:
Yup, overthought too much.

bbb195f4b4ead02032bf4060f900f444698d1c7c